src/avro_record.erl

```diff
@@ -154,6 +155,23 @@ get_all_field_types(Type) when ?IS_RECORD_TYPE(Type) ->
       {FieldName, FieldTypeOrName}
     end, Fields).
 
+%% @doc Get all field types, together with their defaults
+%% in a triple list.
+%% @end
+-spec get_all_field_data(avro_type()) -> [ {field_name()
+                                         , type_or_name(),
+                                         ?NO_VALUE | avro:in() | avro_value()
+                                         }].
+get_all_field_data(Type) when ?IS_RECORD_TYPE(Type) ->
```

Comment:
`?IS_RECORD_TYPE` looks excessive, because it just checks that `Type` is record `#avro_record_type{}` (and you do this on the next line by `#avro_record_type{fields = Fields} = Type`.
Anyway, I personally would code this as `get_all_field_data(#avro_record_type{fields = Fields}) -> ...`.